src/zkevm_specs/evm_circuit/execution/end_tx.py

```diff
@@ -23,22 +23,26 @@ def end_tx(instruction: Instruction):
         instruction.constrain_zero(effective_refund)
 
     # Add effective_refund * gas_price back to caller's balance
-    tx_gas_price = instruction.tx_gas_price(tx_id)
+    tx_gas_fee_cap = instruction.tx_gas_fee_cap(tx_id)
     value, carry = instruction.mul_word_by_u64(
-        tx_gas_price, instruction.curr.gas_left + effective_refund
+        tx_gas_fee_cap, instruction.curr.gas_left + effective_refund
     )
     instruction.constrain_zero(carry)
     tx_caller_address = instruction.tx_context_lookup(tx_id, TxContextFieldTag.CallerAddress)
     instruction.add_balance(tx_caller_address, [value])
 
     # Add gas_used * effective_tip to coinbase's balance
     base_fee = instruction.block_context_lookup_word(BlockContextFieldTag.BaseFee)
-    effective_tip, _ = instruction.sub_word(tx_gas_price, base_fee)
+    effective_tip, _ = instruction.sub_word(tx_gas_fee_cap, base_fee)
```

Comment:
I am wondering that as this is a cap, the real fee should <= than this, so you may not subtract this cap value but another gas value instead. Please double check.

Reply:
No, the `cap` here means `gas_price == min(gas_tip_cap, gas_fee_cap - base_fee)`.  It likes that the gas tip cann't large than gas fee subtract base fee